### ISSUE TITLE
browser windows have their URL's loaded synchronously

### ISSUE THREAD

**Issue by lukeapage** [atom/electron#4143]
I have some code that restores a number of windows to their original position. Something like..

```
for(let i = 0; i < 5; i++) {
    const window = new BrowserWindow({ show: false });
    window.loadURL('http://localhost/url');
    window.setPosition(i * 20, i * 20);
    window.show();
}
```

The windows all show themselves very quickly, but the 1st window loads and only after dom ready on the 1st window has loaded, does the 2nd window start loading, then after that's loaded the 3rd window etc.

I've added logging and the loadURL is not paused, so something else must be making windows wait and load in order..

Would it be difficult to get these loading all at the same time? Any pointers or reasons why it is behaving like this?

Thanks in advance,

**Comment by zcbenz** [comment]
Electron is just calling the API provided by Chromium, I guess it is a feature that window loading is synchronous, so Chromium won't start lots of connections at the same time.

I don't think there is a way to make loadURL async, Chromium doesn't provide related APIs. I'm closing this as won't fix since it is probably very hard to solve and doesn't worth the effort.
